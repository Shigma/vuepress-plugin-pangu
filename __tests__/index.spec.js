/**
 * @jest-environment node
 */

const { fs, path, parseFrontmatter } = require('@vuepress/shared-utils')
const { createApp } = require('@vuepress/core')

describe('spacing', () => {
  let app

  beforeAll(async () => {
    app = createApp({
      siteConfig: {
        locales: {
          '/zh/': { lang: 'zh-CN' },
          '/en/': { lang: 'en-US' },
        },
        plugins: [
          [require('..'), {
          }],
        ],
      },
    })
    return app.process()
  })

  const fragmentDir = path.join(__dirname, 'fragments')
  fs.readdirSync(fragmentDir).forEach((name) => {
    const filepath = path.join(fragmentDir, name)
    const rawFile = fs.readFileSync(filepath, 'utf8')
    const { data, content } = parseFrontmatter(rawFile)
    test(name, () => {
      const { html } = app.markdown.render(content, {
        ...(data.ENV || {}),
        frontmatter: data,
      })
      expect(html).toMatchSnapshot()
    })
  })
})
