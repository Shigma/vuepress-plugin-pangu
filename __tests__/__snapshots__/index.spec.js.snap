// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spacing inline-code.md 1`] = `<p>demo <code>code</code> demo</p>`;

exports[`spacing insert-space.md 1`] = `
<p>中文 English</p>
<p>English 中文</p>
<p>中<em>文 Eng</em>lish</p>
<p>Eng<em>lish 中</em>文</p>
<p>中文 <em>English</em></p>
<p><em>English</em> 中文</p>
<p><em>中文</em> English</p>
<p>English <em>中文</em></p>
<p><em>中文</em> <s><strong>Eng</strong>lish</s></p>
<p><em>中<strong>文</strong></em> <s>English</s></p>
<p>中文 English</p>
<p>English 中文</p>
<p>中<em>文 Eng</em>lish</p>
<p>Eng<em>lish 中</em>文</p>
<p>中文<em>English</em></p>
<p><em>English</em> 中文</p>
<p><em>中文</em> English</p>
<p>English<em>中文</em></p>
<p><em>中文</em><s><strong>Eng</strong>lish</s></p>
<p><em>中<strong>文</strong></em><s>English</s></p>
`;

exports[`spacing punctuation-en.md 1`] = `
<p>大家好, 我是作者.</p>
<p>大家好, 我是作者.</p>
<p>大家好 a,b 我是作者.</p>
<p>大家好 a, b 我是作者.</p>
`;

exports[`spacing punctuation-zh.md 1`] = `
<p>大家好，我是作者。</p>
<p>大家好，我是作者。</p>
<p>大家好 a,b 我是作者。</p>
<p>大家好 a，b 我是作者。</p>
`;
